integration_tests/src/main/python/window_function_test.py

```diff
@@ -444,3 +445,53 @@ def test_window_aggs_for_rows_collect_list():
             (partition by a order by b,c_int rows between CURRENT ROW and UNBOUNDED FOLLOWING) as collect_struct
         from window_collect_table
         ''')
+
+
+# Generates some repeated values to test the deduplication of GpuCollectSet.
+# And GpuCollectSet does not yet support struct type.
+_gen_data_for_collect_set = [
+    ('a', RepeatSeqGen(LongGen(), length=20)),
+    ('b', IntegerGen()),
+    ('c_int', RepeatSeqGen(IntegerGen(), length=15)),
+    ('c_long', RepeatSeqGen(LongGen(), length=15)),
+    ('c_time', RepeatSeqGen(DateGen(), length=15)),
+    ('c_string', RepeatSeqGen(StringGen(), length=15)),
+    ('c_float', RepeatSeqGen(FloatGen(), length=15)),
+    ('c_decimal', RepeatSeqGen(DecimalGen(precision=8, scale=3), length=15)),
+]
+
+
+# Uses sort_array function to reorder the outputs of collect_set, to make the result deterministic.
+@allow_non_gpu('ProjectExec', 'Alias', 'Literal', 'SortArray')
```

Comment:
Can you file a follow on issue to support `sort_array`? In it's most basic form we should be able to support it.  For more complicated ones with actual higher order functions it is going to be a bit more difficult.

Reply:
Done.

Reply:
https://github.com/rapidsai/cudf/blob/48a2663f11b3c36ebed1b4300307cbbd6375c6ca/cpp/include/cudf/lists/sorting.hpp#L51

It is not perfect because Spark uses a higher order function for this, but we can support the default ordering at least, and possibly use pattern matching to figure out other pieces.